include/ccf/tx_id.h

```diff
@@ -64,7 +64,7 @@ namespace ccf
         const auto view_sv = sv.substr(0, separator_idx);
         const auto [p, ec] =
           std::from_chars(view_sv.begin(), view_sv.end(), tx_id.view);
-        if (ec != std::errc() || p != view_sv.end())
+        if (ec != std::errc() || p != view_sv.end() || tx_id.view < 2)
```

Comment:
```suggestion
        if (ec != std::errc() || p != view_sv.end() || tx_id.view == VIEW_UNKNOWN)
```

I don't think we should codify anywhere that `1` is an invalid/illegal/view. It's an incidental implementation detail that `1` is an _empty_ view on current CCF services, but I think we could change/fix that at any point, and shouldn't introduce stronger claims about it. In particular, for the current use case, `1.1` should be a valid TxID, with `TxStatus::Invalid` (because `2.1` is committed).

Reply:
As commented above, we have previously effectively decided that `1.1` could be, but isn't a valid TxID, whereas `0.0` is simply malformed. I don't know that this was a particularly good decision, but staying compatible makes sense (continue to return 200/Invalid and avoid breaking anyone depending on that), what I want to fix is returning a 500 without detail for `0.0`, where 400/NotAValidTxID is clearly preferable and should not break any existing usage.